updater/lib/dependabot/dependency_snapshot.rb

```diff
@@ -220,5 +230,12 @@ def dependency_file_parser
         options: job.experiments
       )
     end
+
+    sig { params(group: Dependabot::DependencyGroup).returns(T::Array[T::Hash[String, String]]) }
+    def dependencies_in_existing_pr_for_group(group)
+      job.existing_group_pull_requests.find do |pr|
+        pr["dependency-group-name"] == group.name
```

Comment:
Any reason why these are not constants specified somewhere to avoid off-by-one or transposes?

Reply:
We should really be parsing these Hashes into proper class instances in job.rb, like mentioned in https://github.com/dependabot/dependabot-core/issues/9561 for ignore conditions.